pkg/util/ansible_execution.go

```diff
@@ -71,7 +71,7 @@ func AnsibleExecution(ctx context.Context, helper *helper.Helper, obj client.Obj
 					SecretName: sshKeySecret,
 					Items: []corev1.KeyToPath{
 						{
-							Key:  "ssh_private_key",
+							Key:  "ssh-privatekey",
```

Comment:
currenlty im am adding both the ssh privatekey and the corresponding public key.

if we wanted to use the same secret with the baremetal operator we would probably want to require both filed are present in the secret.

i think that in general the operator is missing input validation for the secret.

we should probably move this to libcommon eventually 

https://github.com/openstack-k8s-operators/nova-operator/blob/master/controllers/common.go#L82-L147

https://github.com/openstack-k8s-operators/nova-operator/blob/1b8be195d2b37cc93ec33a63cb02e7ea507fbd83/controllers/novaapi_controller.go#L150-L167

but you probably need to add input validation that ensure the required keys are present and report this in the status when not.

Reply:
if we want to declar it out of scope fo this PR and do it in a different one that is fine but we should never get as far as invoking the AnsibleEE runner container before we have validated the input of the secret that we are using.

Reply:
+1 to moving to lib-common

Reply:
I assume we'll merge this with the validation in this repo's common.go and then in a future PR add the key validation to lib-common and update this patch to use it.

Reply:
I started this: https://github.com/openstack-k8s-operators/lib-common/pull/201

Reply:
@SeanMooney the lib-common PR will need more discussion, could you please review this PR as is?
I'll do a follow up PR once the lib-common PR is merged